core/lib/zksync_core/src/state_keeper/batch_executor/mod.rs

```diff
@@ -395,6 +395,10 @@ impl BatchExecutor {
 
         let tx_metrics = ExecutionMetricsForCriteria::new(Some(tx), &tx_result);
 
+        if !vm.has_enough_gas_for_batch_tip() {
+            return TxExecutionResult::BootloaderOutOfGasForBlockTip;
+        }
+
         let (bootloader_dry_run_result, bootloader_dry_run_metrics) = self.dryrun_block_tip(vm);
```

Comment:
do we need to run dryrun?

Reply:
We can remove it, but since it is relatively lightweight, maybe it makes sense to just keep it there

Reply:
Discussed in private, we can do it in a separate PR